src/App.vue

```diff
@@ -325,31 +321,39 @@ export default class App extends Mixins(BcolMixin, DateMixin, FilingTemplateMixi
       // reset errors in case this method is invoked more than once (ie, retry)
       this.resetFlags()
 
-      // get business ID query param
-      const businessId = this.$route?.query?.businessId as string
-      if (!businessId) {
-        this.fetchErrorDialog = true
-        if (!businessId) throw new Error('Invalid business identifier')
-      }
-      this.setBusinessId(businessId)
-
       // ensure user is authorized or is staff to access this business
-      await this.checkAuth(businessId).catch(error => {
+      await this.checkAuth(this.getBusinessId).catch(error => {
         this.accountAuthorizationDialog = true
         throw new Error(`Auth error: ${error}`)
       })
 
-      // fetch IA filing to alter or correct
-      const { filing } = await this.fetchFiling(this.INCORPORATION_APPLICATION) // TEMP FOR TESTING
-      // const { filing } = await this.fetchFiling(this.ALTERATION) // FUTURE STATE
-
-      // parse filing into store
-      this.parseIncorpFiling(filing)
+      // check for necessary query params
+      switch (this.$route.name) {
+        case RouteNames.ALTERATION:
+          // const filingId = +this.$route.query['filing-id']
+          // if (isNaN(filingId)) {
+          //   this.fetchErrorDialog = true
+          //   throw new Error('Invalid alteration filing ID')
+          // }
+          this.setEntityType(EntityTypes.BCOMP)
+          break
+        case RouteNames.CORRECTION:
+          const correctedId = +this.$route.query['corrected-id']
```

Comment:
The Internet consensus is to use spinal-case for URI/URL stuff.
https://blog.restcase.com/5-basic-rest-api-design-guidelines/
https://stackoverflow.com/questions/10302179/hyphen-underscore-or-camelcase-as-word-delimiter-in-uris

Reply:
Interesting, i guess there is a protocol for everything!